src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js

```diff
@@ -1,128 +1,410 @@
+                    <ExpandableRowContent>{ rowData.data.modules.filter(a => a).join(', ') }</ExpandableRowContent>
+                ) : (
+                    value
+                ),
+            renderEdit: (value, { column, columnIndex, rowIndex, rowData }, { activeEditId }) => {
+                const renderer = rowData.hasOwnProperty('parent') ? childEditRenderer : parentEditRenderer;
+                return renderer(value, { rowData, column, rowIndex, columnIndex, activeEditId });
+            }
+        });
 
-        if (!baselineData) {
-            rows = [];
+        /*eslint-disable react/jsx-key*/
+        let rows = [ ...new Array(5) ].map(() =>
+            [ ...new Array(2) ].map(() =>
+                <Skeleton size={ SkeletonSize.md } />
+            )
+        );
+        /*eslint-enable react/jsx-key*/
+
+        let displayName;
+
+        if (this.props.baselineData) {
+            displayName = this.props.baselineData.display_name;
         } else {
-            baselineData.baseline_facts.forEach(function(fact) {
-                let row = [];
-                row.push(fact.name);
-                row.push(fact.value);
-                rows.push(row);
-            });
+            displayName = '';
         }
 
-        return rows;
-    }
+        this.state = {
+            columns: [ 'Fact', 'Value' ],
+            rows,
+            editedRowsBackup: null,
+            activeEditId: null,
+            baselineName: displayName
+        };
 
-    renderLoadingRows() {
-        let rows = [];
-        let rowData = [];
+        this.changeBaselineName = value => {
+            this.setState({ value });
+        };
 
-        for (let i = 0; i < 2; i += 1) {
-            rowData.push(<Skeleton size={ SkeletonSize.md } />);
-        }
+        this.submitBaselineName = (/*baselineName*/) => {
+        };
 
-        for (let i = 0; i < 5; i += 1) {
-            rows.push(rowData);
-        }
+        this.onChange = (value, { rowIndex, columnIndex, moduleIndex }) => {
+            this.setState(({ rows }) => {
+                const { baselineData } = this.props;
+                rows = [ ...rows ];
+                const row = rows[rowIndex];
+                if (moduleIndex !== null && moduleIndex !== undefined) {
+                    row.data.modules[moduleIndex] = value;
+                } else {
+                    const shiftedColumnIndex = columnIndex - 1; // to take Expandable Column into account;
+                    row.cells[shiftedColumnIndex] = value;
+                }
+
+                let apiBody = this.buildApiBody(rows, rowIndex);
+                api.patchBaselineData(baselineData.id, apiBody);
+
+                return { rows, activeEditId: null };
+            });
+        };
+
+        this.buildApiBody = (rows, rowIndex) => {
+            let row = rows[rowIndex];
+            let apiBody = {};
+            let baselineFactsArray = [];
+
+            /*eslint-disable camelcase*/
```

Comment:
can the camelcase be rewritten with `_`'s? parent_row for example

Reply:
This disables the requirement of camelcase. It's because of baseline_facts.